Job Post Analysis and Report

In [13]:
# Import General Libraries
import pandas as pd 
import numpy as np 

# Import Visualization Libraries
import matplotlib.pyplot as plt 
import seaborn as sns 

## Loading Data

In [14]:
df = pd.read_csv(r'transformed_jobs_data.csv')
df.head(3)

job_id job_employment_type  \
0  XX6946dvNO3187IkAAAAAA==            FULLTIME   
1  6Wu7QcWfrhOR-THoAAAAAA==            FULLTIME   
2  Oiu186OT5E6bkZReAAAAAA==            FULLTIME   

                                           job_title  \
0                                      Data Engineer   
1                                      Data Engineer   
2  2024 RBCIS, Summer Co-op Data Engineer / Full ...   

                                      job_apply_link  \
0  https://ca.linkedin.com/jobs/view/data-enginee...   
1  https://ca.linkedin.com/jobs/view/data-enginee...   
2  https://ca.linkedin.com/jobs/view/2024-rbcis-s...   

                                     job_description  job_city job_country  \
0  • *Please note that our offices will be closed...  Montréal          CA   
1  We are seeking an experienced and passionate D...       NaN          CA   
2  Job Summary\n\nJob Description\n\nWhat is the ...   Toronto          CA   

  job_posted_at_timestamp    employer_website employer_company_type  
0     2023-12-19 20:22:04                 NaN                   NaN  
1     2023-12-19 15:55:49                 NaN                   NaN  
2     2023-12-19 22:30:06  http://www.rbc.com               Finance

In [15]:
# df['job_city'].unique()

In [16]:
df.tail()

job_id job_employment_type                   job_title  \
161  0JxLam0cJtXsmjQZAAAAAA==            FULLTIME  Senior Cloud Data Engineer   
162  EGvoxwF7K6W0LOMtAAAAAA==            FULLTIME  Senior Cloud Data Engineer   
163  TvhAbNPDfD24fyHVAAAAAA==            FULLTIME  Senior Cloud Data Engineer   
164  6N_3eAHv_8HXYM0cAAAAAA==            FULLTIME  Senior Cloud Data Engineer   
165  iQu2KNPFHbJstSZuAAAAAA==            FULLTIME  Senior Cloud Data Engineer   

                                        job_apply_link  \
161  https://www.linkedin.com/jobs/view/senior-clou...   
162  https://www.linkedin.com/jobs/view/senior-clou...   
163  https://www.linkedin.com/jobs/view/senior-clou...   
164  https://www.linkedin.com/jobs/view/senior-clou...   
165  https://www.linkedin.com/jobs/view/senior-clou...   

                                       job_description         job_city  \
161  Job Description\n\nJob Summary:\n\nThis positi...        Las Vegas   
162  Job Description\n\nJob Summary:\n\nThis positi...         Richmond   
163  Job Description\n\nJob Summary:\n\nThis positi...       Fort Worth   
164  Job Description\n\nJob Summary:\n\nThis positi...         Columbus   
165  Job Description\n\nJob Summary:\n\nThis positi...  Fort Lauderdale   

    job_country job_posted_at_timestamp    employer_website  \
161          US     2023-12-19 12:11:14  http://www.bdo.com   
162          US     2023-12-19 10:42:36  http://www.bdo.com   
163          US     2023-12-19 11:52:55  http://www.bdo.com   
164          US     2023-12-19 12:15:04  http://www.bdo.com   
165          US     2023-12-19 12:11:14  http://www.bdo.com   

    employer_company_type  
161            Consulting  
162            Consulting  
163            Consulting  
164            Consulting  
165            Consulting

In [17]:
df.shape

(166, 10)

In [18]:
df.isnull().sum()

job_id                      0
job_employment_type         0
job_title                   0
job_apply_link              0
job_description             0
job_city                   38
job_country                 0
job_posted_at_timestamp     0
employer_website           53
employer_company_type      90
dtype: int64

In [19]:
# Function to format null values to 2dp
def formatt(x):
    return '{:.2f}%'.format(x)

((df.isnull().sum() / len(df)) * 100).apply(formatt)

job_id                      0.00%
job_employment_type         0.00%
job_title                   0.00%
job_apply_link              0.00%
job_description             0.00%
job_city                   22.89%
job_country                 0.00%
job_posted_at_timestamp     0.00%
employer_website           31.93%
employer_company_type      54.22%
dtype: object

- Presence of missing data in **THREE** columns namely:
    - job_city [22.89%]
    - employer_website [31.93%]
    - employer_company_type [54.22%]

In [20]:
df.nunique()

job_id                     166
job_employment_type          4
job_title                   99
job_apply_link             166
job_description            153
job_city                    70
job_country                  3
job_posted_at_timestamp    154
employer_website            88
employer_company_type       13
dtype: int64

In [21]:
df.duplicated().value_counts()

False    166
Name: count, dtype: int64

- No duplicates.

In [22]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 166 entries, 0 to 165
Data columns (total 10 columns):
 #   Column                   Non-Null Count  Dtype 
---  ------                   --------------  ----- 
 0   job_id                   166 non-null    object
 1   job_employment_type      166 non-null    object
 2   job_title                166 non-null    object
 3   job_apply_link           166 non-null    object
 4   job_description          166 non-null    object
 5   job_city                 128 non-null    object
 6   job_country              166 non-null    object
 7   job_posted_at_timestamp  166 non-null    object
 8   employer_website         113 non-null    object
 9   employer_company_type    76 non-null     object
dtypes: object(10)
memory usage: 13.1+ KB


In [23]:
# Impute missing job_city 
modeCity = df.groupby('job_country')['job_city'].count().reset_index()
modeCity.columns = ['job_country', 'job_city']
modeCity

    

job_country  job_city
0          CA        26
1          GB        48
2          US        54

In [30]:
# Function to fill missing volumes with the mode value of each city
# def fill_missing_city(df):
#     if np.isnan(df['job_city']):
#         return modeCity[modeCity['job_country'] == df['job_country']]['mode']
    
#     return df['job_city']

# df['job_city'] = df.apply(fill_missing_city, axis=1)

In [39]:
df.columns

Index(['job_id', 'job_employment_type', 'job_title', 'job_apply_link',
       'job_description', 'job_city', 'job_country', 'job_posted_at_timestamp',
       'employer_website', 'employer_company_type'],
      dtype='object')

In [40]:
from sklearn.experimental import enable_iterative_imputer 
from sklearn.impute import IterativeImputer

In [41]:
city_imputer = IterativeImputer(max_iter=10, random_state=0)
city_imputer.fit([[]])

In [ ]:
len(df['job_city'].unique())

71

In [ ]:
# imputer = SimpleImputer(strategy='most_frequent')
# imputer.fit_transform(df['job_city'])

In [36]:
# Impute job_city using the most frequent value
# from sklearn.impute import SimpleImputer
# impute_mode = SimpleImputer(strategy='most_frequent')
# impute_mode.fit(df[['job_city']])

# df[['job_city']] = impute_mode.fit_transform(df[['job_city']].values.reshape(-1,1))
# df['job_city'].isna().sum()


0

In [37]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 166 entries, 0 to 165
Data columns (total 10 columns):
 #   Column                   Non-Null Count  Dtype 
---  ------                   --------------  ----- 
 0   job_id                   166 non-null    object
 1   job_employment_type      166 non-null    object
 2   job_title                166 non-null    object
 3   job_apply_link           166 non-null    object
 4   job_description          166 non-null    object
 5   job_city                 166 non-null    object
 6   job_country              166 non-null    object
 7   job_posted_at_timestamp  166 non-null    object
 8   employer_website         113 non-null    object
 9   employer_company_type    76 non-null     object
dtypes: object(10)
memory usage: 13.1+ KB


In [33]:
# Drop missing rows
# df_copy = df.copy()

# df_copy.dropna(inplace=True)

# df_copy.isna().sum()

### Exploratory Data Analysis